evm/src/cpu/kernel/asm/sha2/memory.asm

```diff
@@ -0,0 +1,162 @@
+// Load a single byte from kernel general memory.
+%macro mload_kernel_general
+    // stack: offset
+    PUSH @SEGMENT_KERNEL_GENERAL
+    // stack: segment, offset
+    PUSH 0 // kernel has context 0
+    // stack: context, segment, offset
+    MLOAD_GENERAL
+    // stack: value
+%endmacro
+
+// Load a big-endian u32, consisting of 4 bytes (c_3, c_2, c_1, c_0),
+// from kernel general memory.
+%macro mload_kernel_general_u32
```

Comment:
Let's add a `%macro mload_kernel_u32(segment)`, so this and `mload_kernel_code_u32` can share (source) code.